### PR TITLE
[SYCL][NFC] Use relative paths for SYCL headers install

### PR DESCRIPTION
**Pull request by npmiller**
`${SYCL_INCLUDE_DEPLOY_DIR}` is defined as
`${CMAKE_INSTALL_PREFIX}/${SYCL_INCLUDE_DIR}`, this is unnecessary
because when using relative paths, CMake will automatically prefix
install destinations with `${CMAKE_INSTALL_PREFIX}`, so we can directly
use `${SYCL_INCLUDE_DIR}`.

It is also preferable to use relative paths, because it enables the
`DESTDIR` mechanism that can make it possible to change the install
prefix after the configuration, with for example things like `make
DESTDIR=/new/install/prefix install`, or when manually calling the CMake
install scripts.

This patch has no functional changes and should have no effect on
existing builds, aside from enabling `DESTDIR`.

### CODE REVIEW

File: sycl/CMakeLists.txt [hunk 1]
```diff
@@ -85,7 +85,7 @@ file(COPY ${OpenCL_INCLUDE_DIR}/CL
 
 # Include OpenCL Headers into final bundle.
 install(DIRECTORY ${OpenCL_INCLUDE_DIR}/CL
-        DESTINATION ${SYCL_INCLUDE_DEPLOY_DIR}/sycl
```
Comment (dm-vodopyanov) [inline sycl/CMakeLists.txt:88]: @npmiller, can you please remove initialization of `SYCL_INCLUDE_DEPLOY_DIR` variable and replace it with `${CMAKE_INSTALL_PREFIX}/${SYCL_INCLUDE_DIR}` in `sycl/test/CMakeLists.txt`?

**Review by dm-vodopyanov — COMMENTED** [review]
Other part LGTM. Thanks for the patch!

File: sycl/CMakeLists.txt (hunk 1 above)
Comment (npmiller) [reply sycl/CMakeLists.txt:88]: Sure, patch updated!

**Review by dm-vodopyanov — APPROVED** [review]
Thanks, LGTM. Please avoid force-pushing in the future, it makes the reviewing difficult as the info about changes disappears.

**Pull request merged**